packages/workbox-background-sync/src/lib/request-queue.js

```diff
@@ -117,31 +126,42 @@ class RequestQueue {
         id: hash,
         url: request.url,
       });
-    } catch(e) {
+
+      return hash;
+    } catch (err) {
       // broadcast the failure of request added to the queue
       broadcastMessage({
         broadcastChannel: this._broadcastChannel,
         type: broadcastMessageFailedType,
         id: hash,
         url: request.url,
       });
+
+      return err;
     }
   }
 
   /**
    * get the Request from the queue at a particular index
    *
    * @param {string} hash hash of the request at the given index
-   * @return {Request} request object corresponding to given hash
+   * @return {Promise} request object corresponding to given hash
```

Comment:
This doesn't fulfill to a `Request` instance, it fulfills to an object with `config`, `metadata`, and `request` properties, and the `request` property is a JSON-able version of the `Request` object.

I could update it to:

```js
/** @return {Promise<Object>} */
```

But I didn't see much of that in the existing code.